sources/PTYSession.m

```diff
@@ -7556,6 +7558,24 @@ - (void)mitigateTouchBarStupidityForEvent:(NSEvent *)event {
     [self showVisualIndicatorForEvent:event];
 }
 
+- (void)selectionDidChange:(NSString *)selection {
+    // Assign a maximum of maximumBytesToProvideToPythonAPI characters to the "selection"
+    // iTerm Variable
+    //
+    // The "selectionLength" iTerm variable contains the full length of the original
+    // selection; not the restricted length assigned to the "selection" iTerm Variable
+    DLog(@"selectionDidChange to %@", selection);
+    
+    if (selection && selection.length > 0) {
+        const int maxLength = [iTermAdvancedSettingsModel maximumBytesToProvideToPythonAPI];
+        [self.variablesScope setValue:[selection substringToIndex:MIN(maxLength, selection.length)] forVariableNamed:iTermVariableKeySessionSelection];
+        [self.variablesScope setValue:@(selection.length) forVariableNamed:iTermVariableKeySessionSelectionLength];
+    } else {
```

Comment:
If we guarantee that `selection` is nonnull then the else clause can be eliminated.

Reply:
I think that rather than ensuring this delegate is never called with a nil – which is impossible in ObjC anyway – we could accept the nil as an indicator of no selection.

We can then use `selection = selection ? selection : @"";` towards the start of the delegate to ensure that a valid `NSString` is available when the iTerm variables are set – allowing the for the removal of the if/else and the almost-duplicate `setValue` lines.  Seem OK?